### ISSUE TITLE
`.over` no longer allowed inside `.arr.eval`

### ISSUE THREAD

**Issue by cmdlineluser** [pola-rs/polars#8721]
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

I was trying the code from https://stackoverflow.com/questions/72871905/polars-intersection-of-list-columns-in-dataframe mentioned in issue #3905 

It now raises an exception:
```
InvalidOperationError: window expression not allowed in aggregation
```
I guess it's due to: [perf(rust, python): arr.eval run on groupby expression engine when possible](https://github.com/pola-rs/polars/pull/8199)

### Reproducible example

```python
import polars as pl

pl.Series([[1, 2, 3, 4]]).arr.eval(pl.count().over(pl.element()))
```

### Expected behavior

I'm not sure. 

I see #6676 exists regarding allowing window expressions inside aggregation.

Perhaps if the `.over` is present, this can be detected and the expression is not allowed to run on the groupby engine?

### Installed versions

<details>

```
--------Version info---------
Polars:      0.17.12
Index type:  UInt32
Python:      3.10.9 (main, Jan 14 2023, 16:56:26)

----Optional dependencies----
numpy:       1.23.5
pandas:      2.0.0
pyarrow:     9.0.0
connectorx:  <not installed>
deltalake:   <not installed>
fsspec:      2022.8.2
matplotlib:  3.5.1
xlsx2csv:    0.8
xlsxwriter:  3.0.8
```

</details>

**Comment by ritchie46** [comment]
Yes, this is due to that. Now I am really wondering if we should allow this. Otherwise we would do a window function per element in the list which is insanely expensive.
